### PR TITLE
Added linter setup and fixed all current errors

### PR DESCRIPTION
**Pull request by CptPie**
For linting [golangci-lint](https://golangci-lint.run/) was used, a corresponding configuration is provided.

A few linter settings are to be discussed but i think this is a good basis.

### CODE REVIEW

**Comment by CptPie** [comment]
For now this linting is neither enforced nor integrated into the build process.

File: logic/security.go [hunk 1]
```diff
@@ -63,19 +63,6 @@ func (b *backend) HashPassword(pass string) string {
 	return fmt.Sprintf("%x", sha512.Sum512([]byte(b.passwordSalt+pass)))
 }
 
-func (b *backend) generatePass() (string, error) {
```
Comment (joeyak) [inline logic/security.go:66]: RIP

File: web/pageAdmin.go [hunk 2]
```diff
@@ -165,7 +165,16 @@ func (s *webServer) handlerAdminUserEdit(w http.ResponseWriter, r *http.Request)
 
 		return
 	case "ban":
-		s.backend.AdminBanUser(user)
+		err = s.backend.AdminBanUser(user)
+		if err != nil {
+			if err != nil {
```
Comment (joeyak) [inline web/pageAdmin.go:170]: Double err checks here

**Review by joeyak — APPROVED** [review]
Besides the double err check, everything looks good. I'm guessing the double check will be handled in the error rework

File: web/pageAdmin.go (hunk 2 above)
Comment (CptPie) [reply web/pageAdmin.go:170]: That was just me being stupid lul, i'll fix that tomorrow

Comment (zorchenhimer) [reply web/pageAdmin.go:170]: Bumping this so i can merge this hopefully soon.

**Review by zorchenhimer — CHANGES_REQUESTED** [review]

**Review by zorchenhimer — APPROVED** [review]

**Pull request merged**